data/scripts.js

```diff
@@ -1702,6 +1702,8 @@ exports.BattleScripts = {
 				rejectAbility = !counter['Bug'];
 			} else if (ability === 'Technician') {
 				rejectAbility = !counter['technician'] || (abilities.indexOf('Skill Link') >= 0 && counter['skilllink'] >= counter['technician']);
+			} else if (ability === 'Tinted Lens') {
+				rejectAbility = counter['damage'] >= counter.damagingMoves.length;
```

Comment:
Fixed damage attacks like Night Shade are also included in that array, and the other counter we compare to tracks those kinds of moves.